### ISSUE TITLE
Error: Match error: Expected string, got null

### ISSUE THREAD

**Issue by patrickbolle** [ARKHAM-Enterprises/meteor-comments-ui#94]
Hey there,
I've got this half working...

I just put in your 
```
    <div class="comment-section">
    {{> commentsBox id=post._id}}
    </div>
```
code from the main page.
In console I get this error over and over again - 
`Exception while invoking method 'comments/count' Error: Match error: Expected string, got null
I20160623-13:43:24.992(-4)?     at exports.check (packages/check/match.js:34:1)
I20160623-13:4`

For count/add/reference 

I haven't touched anything besides putting that comment-section into my code, I'm on the newest release of Meteor.

Thanks!

**Comment by matteodem** [comment]
can you submit a small reproduction repository? Just in case that I cannot reproduce it locally.

**Comment by patrickbolle** [comment]
Hey!
Here is the full repository - https://github.com/patrickbolle/reimagined-goggles

Here is the file where I drop the comments template - 
https://github.com/patrickbolle/reimagined-goggles/blob/master/client/post/postDetail.html

(Don't mind the other 'comments' files in the repo, they are just my attempts at doing it by hand).

So when I run the app, the comments work perfectly when I load a 'post'. But when I REFRESH the page, my console gets filled with this error.

Thanks man.

**Comment by patrickbolle** [comment]
Woaah, alright I got it figured out.
I'm using Flow Router as opposed to Iron Router, and I guess I was forgetting template level subscriptions (Iron router I would have just done a waitOn in the route).
So I added this to my "postDetail" template js
```
Template.postDetail.onCreated(function() {
  var self = this;
  self.autorun(function() {
    var id = FlowRouter.getParam('id');
    self.subscribe('Posts', id);  
  });
});
```

Then wrapped my html template with this
```
    {{#if Template.subscriptionsReady}}
      {{#with post}}
```

Boom! Now it works beautifully, well except I need to change the UI as I'm using Picnic css.
This also fixed a similar problem I had with the stars-rating-ui package!

Thanks!
